src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -63,35 +62,40 @@
 
 import jenkins.model.Jenkins;
 
+import jenkins.tasks.SimpleBuildStep;
 import org.acegisecurity.Authentication;
 import org.acegisecurity.GrantedAuthority;
 import org.acegisecurity.providers.UsernamePasswordAuthenticationToken;
 import org.apache.commons.lang.StringUtils;
 import org.kohsuke.stapler.AncestorInPath;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.Stapler;
 import org.kohsuke.stapler.StaplerRequest;
 
+import javax.annotation.Nonnull;
+
 /**
  * Build step to copy artifacts from another project.
  * @author Alan Harder
  */
-public class CopyArtifact extends Builder {
+public class CopyArtifact extends Builder implements SimpleBuildStep {
 
     // specifies upgradeCopyArtifact is needed to work.
     private static boolean upgradeNeeded = false;
     private static Logger LOGGER = Logger.getLogger(CopyArtifact.class.getName());
+    private static final BuildSelector DEFAULT_BUILD_SELECTOR = new StatusBuildSelector(true);
 
     @Deprecated private String projectName;
     private String project;
     private String parameters;
-    private final String filter, target;
-    private final String excludes;
+    private String filter, target;
+    private String excludes;
     private /*almost final*/ BuildSelector selector;
     @Deprecated private transient Boolean stable;
-    private final Boolean flatten, optional;
-    private final boolean doNotFingerprintArtifacts;
+    private Boolean flatten, optional;
```

Comment:
I did not make these fields `Boolean`. This is how they were in the original code. I don't see any reason for them not being `boolean` either but would prefer to check with @ikedam before making any changes. Better still, perhaps do it in a separate PR since it's not directly related to what we're trying to do here.